lib/ddtrace/tracer.rb

```diff
@@ -355,10 +358,10 @@ def configure_writer(options = {})
       sampler = options.fetch(:sampler, nil)
       priority_sampling = options.fetch(:priority_sampling, nil)
       writer = options.fetch(:writer, nil)
-      transport_options = options.fetch(:transport_options, {})
+      transport_options = options.fetch(:transport_options, {}).dup
```

Comment:
This was a bug where building options for a new writer would mutate the provided Hash by reference, causing unwanted side effects.